### PR TITLE
Create non-existing binding volumes

### LINKED ISSUE #185: Create non-existing bind volumes
I am not sure whether `docker` or `docker-compose` does this and conversely I don't know whether the issue lies with `podman` or `podman-compose`:

For postgres containers we often have a setup like this:

```
services:

  db:
    image: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: example
    volumes:
    - "./pg-data:/var/lib/postgresql/data"
```

in docker the folder `pg-data` is created automatically, if it doesn't exist. Via podman-compose I get

```
Error: error checking path "/tmp/foo/./pg-data": stat /tmp/foo/./pg-data: no such file or directory
125
```

### PR DESCRIPTION
**Pull request by lcarva**
Ideally, this would be done by podman, but in the mean time do it in
podman-compose for a better user experience.

Fixes #185

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

File: podman_compose.py [hunk 1]
```diff
@@ -89,6 +89,8 @@ def parse_short_mount(mount_str, basedir):
         mount_type = "bind"
         # TODO: should we use os.path.realpath(basedir)?
         mount_src = os.path.join(basedir, os.path.expanduser(mount_src))
+        if not os.path.exists(mount_src):
```
Comment (xflavioribeiro) [inline podman_compose.py:92]: use `try/catch` instead a conditional prevent a unexpected issue

Comment (xflavioribeiro) [inline podman_compose.py:92]: use `try/catch` instead a conditional prevent a unexpected issue

Comment (lcarva) [reply podman_compose.py]: Good idea! Fixed it.

**Comment by lcarva** [comment]
> Them user could mount in a non permitted directory and cause a issue

If that's the case, do we want the exception to bubble up? Or silently fail? Bubbling up the error might give the user a better idea of what the problem actually is.

**Comment by xflavioribeiro** [comment]
Errors like it should be catched and printed to user understand what is wrong.

**Comment by lcarva** [comment]
Updated the PR to only perform this action if in compatibility mode as suggested in #185.

File: podman_compose.py [hunk 2]
```diff
@@ -89,6 +89,13 @@ def parse_short_mount(mount_str, basedir):
         mount_type = "bind"
         # TODO: should we use os.path.realpath(basedir)?
         mount_src = os.path.join(basedir, os.path.expanduser(mount_src))
+        if compose.compatibility_mode:
+            try:
+                os.makedirs(mount_src, exist_ok=True)
```
Comment (martinetd) [inline podman_compose.py:94]: just a nitpick with this: if the bind source is an existing file and not a directory, os.makedirs will fail even with exist_ok, so a previous check is required

Comment (martinetd) [reply podman_compose.py:94]: nm I'm blind it's just below, deleting this comment

**Comment by amn** [comment]
I see that the compatibility mode [is not in the principal (`devel`) branch](https://github.com/containers/podman-compose/blob/devel/podman_compose.py#L388) -- issue #185 was closed but the commit did not introduce the compatibility mode, was the idea scrapped? There's the [PR #194](https://github.com/containers/podman-compose/pull/194) with [a commit adding the mode](https://github.com/containers/podman-compose/pull/194/commits/98bd4cf1ce9f97907b8d608f9d49d95e6b84ec28) but it's not merged. Sorry for sharing my confusion, but I tripped over the "let's create empty directories for volumes for which source files are missing" behaviour today, as I haven't used Docker and am not used to its behaviour.

**Pull request closed without merging**